pkg/trace/agent/normalizer.go

```diff
@@ -158,6 +158,25 @@ func (a *Agent) normalize(ts *info.TagStats, s *pb.Span) error {
 			delete(s.Meta, "http.status_code")
 		}
 	}
+
+	if len(s.SpanLinks) > 0 {
+		validatedLinks := make([]*pb.SpanLink, len(s.SpanLinks))
```

Comment:
Let's try to avoid making a new slice if we can. You can probably just remove in place, if needed. e.g. https://go.dev/play/p/pSwzLfnCbZF

Reply:
To summarize what we discussed offline:
It may be fine to keep span links that have trace_ids or span_ids that are zero if it keeps things simple. We should check with the backend folks though to see if this happens commonly, and what they'll do if they receive a span link with a zero value id.
For the normalization of the `link.name`, we said we'll do the same sanitization as the operation name, and document that publicly.
Still need to figure out `dd.kind`.